### PR TITLE
[fix](cloud) Check instance_id valid when use cloud_unique_id degrade format

### PR DESCRIPTION
**Pull request by deardeng**
### What problem does this PR solve?

Issue Number: close #xxx

Related PR: #xxx

Problem Summary:

### Check List (For Committer)

- Test 

    - [ ] Regression test
    - [x] Unit Test
    - [ ] Manual test (add detailed scripts or steps below)
    - [ ] No need to test or manual test. Explain why:
        - [x] This is a refactor/code format and no logic has been changed.
        - [x] Previous test can cover this change.
        - [x] No colde files have been changed.
        - [ ] Other reason 

- Behavior changed:

    - [x] No.
    - [ ] Yes. 

- Does this need documentation?

    - [x] No.
    - [ ] Yes. 

- Release note

    
    
    None

### Check List (For Reviewer who merge this PR)

- [ ] Confirm the release note
- [ ] Confirm test cases
- [ ] Confirm document
- [ ] Add branch pick label

### CODE REVIEW

**Comment by deardeng** [comment]
run buildall

File: cloud/src/common/config.h [hunk 1]
```diff
@@ -219,4 +219,6 @@ CONF_Int32(max_tablet_index_num_per_batch, "1000");
 
 // Max aborted txn num for the same label name
 CONF_mInt64(max_num_aborted_txn, "100");
+
+CONF_Bool(enable_check_cloud_unique_id_degrade_format, "true");
```
Comment (dataroaring) [inline cloud/src/common/config.h:223]: enable_check_instance_id

File: cloud/src/resource-manager/resource_manager.cpp [hunk 2]
```diff
@@ -199,6 +211,28 @@ bool ResourceManager::check_cluster_params_valid(const ClusterPB& cluster, std::
     return no_err;
 }
 
+std::pair<bool, std::string> ResourceManager::get_instance_id_by_degrade_unique_id(
```
Comment (dataroaring) [inline cloud/src/resource-manager/resource_manager.cpp:214]: get_instance_id_from_unique_id.

File: cloud/src/meta-service/meta_service.cpp [hunk 3]
```diff
@@ -88,46 +88,52 @@ std::string get_instance_id(const std::shared_ptr<ResourceManager>& rc_mgr,
...
-        for (int i = 0; i < vec.size(); ++i) {
-            ss << "idx " << i << "= [" << vec[i] << "] ";
-        }
-        LOG(INFO) << "degraded to get instance_id, cloud_unique_id: " << cloud_unique_id
-                  << "after split: " << ss.str();
-        if (vec.size() != 3) {
-            LOG(WARNING) << "cloud unique id is not degraded format, failed to check instance "
-                            "info, cloud_unique_id="
-                         << cloud_unique_id << " , err=" << err;
+        auto [valid, instance_id] = rc_mgr->get_instance_id_by_degrade_unique_id(cloud_unique_id);
```
Comment (gavinchou) [inline cloud/src/meta-service/meta_service.cpp:96]: dangrous shadow variable

File: cloud/src/resource-manager/resource_manager.h [hunk 4]
```diff
@@ -114,6 +114,25 @@ class ResourceManager {
...
                                     bool check_master_num);
 
+    /**
+     * Check cloud_unique_id is degraded format, and get instance_id from cloud_unique_id
+     * degraded format : "${version}:${instance_id}:${unique_id}"
+     * @param degraded cloud_unique_id
+     *
+     * @return a <is_degraded_format, instance_id> pair, if is_degraded_format == true , instance_id, if is_degraded_format == false, instance_id=""
+     */
+    std::pair<bool, std::string> get_instance_id_by_degrade_unique_id(
```
Comment (gavinchou) [inline cloud/src/resource-manager/resource_manager.h:124]: naming "get_instance_id_by_cloud_unique_id()"
and make it a static member function

File: cloud/src/resource-manager/resource_manager.cpp [hunk 5]
```diff
@@ -199,6 +211,28 @@ bool ResourceManager::check_cluster_params_valid(const ClusterPB& cluster, std::
...
+bool ResourceManager::check_degrade_instance_valid(const std::string& instance_id) {
+    // check kv
+    auto [c0, m0] = get_instance(nullptr, instance_id, nullptr);
+    { TEST_SYNC_POINT_CALLBACK("check_degrade_instance_valid", &c0); }
+    if (c0 != TxnErrorCode::TXN_OK) {
+        LOG(WARNING) << "check instance instance_id=" << instance_id
+                     << " failed, code=" << format_as(c0) << ", info=" + m0;
+        return false;
+    }
+    return true;
```
Comment (gavinchou) [inline cloud/src/resource-manager/resource_manager.cpp:233]:
```suggestion
    if (c0 != TxnErrorCode::TXN_OK) {
        LOG(WARNING) << "failed to check instance instance_id=" << instance_id
                     << " expected= " << ... << " given=" << ...
                     << " failed, code=" << format_as(c0) << ", info=" + m0;
    }
    return c0 == TxnErrorCode::TXN_OK;
```

File: cloud/src/resource-manager/resource_manager.h [hunk 6]
```diff
@@ -114,6 +114,25 @@ class ResourceManager {
...
+            const std::string& cloud_unique_id);
+
+    /**
+     * check instance_id is a valid instance, check by get fdb kv 
+     *
+     * @param instance_id
+     *
+     * @return true, instance_id in fdb kv
+     */
+    bool check_degrade_instance_valid(const std::string& instance_id);
```
Comment (gavinchou) [inline cloud/src/resource-manager/resource_manager.h:134]: naming "is_instance_id_valid()" or "is_instance_id_registered()"

**Review by gavinchou — CHANGES_REQUESTED** [review]

**Comment by deardeng** [comment]
run buildall

File: cloud/src/meta-service/meta_service.cpp [hunk 7]
```diff
@@ -88,46 +88,52 @@ std::string get_instance_id(const std::shared_ptr<ResourceManager>& rc_mgr,
...
-            // just return instance id;
-            return vec[1];
-        default:
-            LOG(WARNING) << "cloud unique id degraded state, but version not eq configure, "
-                            "cloud_unique_id="
-                         << cloud_unique_id << ", err=" << err;
+
+        // check instance_id valid by get fdb
+        if (config::enable_check_instance_id && !rc_mgr->is_instance_id_registered(id)) {
+            LOG(WARNING) << "use degraded format cloud_unique_id, but check instance failed";
```
Comment (gavinchou) [inline cloud/src/meta-service/meta_service.cpp:106]: should log cloud_unique_id and instance id

**Review by gavinchou — DISMISSED** [review]

**Comment by deardeng** [comment]
run buildall

**Comment by deardeng** [comment]
run cloudut

**Comment by deardeng** [comment]
run buildall

**Review by gavinchou — APPROVED** [review]

**Review by dataroaring — APPROVED** [review]
LGTM

**Pull request merged**